internal/langserver/handlers/command/module_calls.go

```diff
@@ -48,72 +49,41 @@ func (h *CmdHandler) ModuleCallsHandler(ctx context.Context, args cmd.CommandArg
 		return response, err
 	}
 
-	found, _ := h.StateStore.Modules.ModuleByPath(modPath)
-	if found == nil {
-		return response, nil
-	}
-
-	if found.ModManifest == nil {
-		return response, nil
+	moduleCalls, err := h.StateStore.Modules.ModuleCalls(modPath)
+	if err != nil {
+		return response, err
 	}
 
-	response.ModuleCalls = h.parseModuleRecords(ctx, found.ModManifest.Records)
+	response.ModuleCalls = h.parseModuleRecords(ctx, moduleCalls)
 
 	return response, nil
 }
 
-func (h *CmdHandler) parseModuleRecords(ctx context.Context, records []datadir.ModuleRecord) []moduleCall {
-	// sort all records by key so that dependent modules are found
-	// after primary modules
-	sort.SliceStable(records, func(i, j int) bool {
-		return records[i].Key < records[j].Key
-	})
-
+func (h *CmdHandler) parseModuleRecords(ctx context.Context, moduleCalls tfmod.ModuleCalls) []moduleCall {
 	modules := make(map[string]moduleCall)
-	for _, manifest := range records {
-		if manifest.IsRoot() {
-			// this is the current directory, which is technically a module
-			// skipping as it's not relevant in the activity bar (yet?)
+	for _, manifest := range moduleCalls.Declared {
+		if manifest.SourceAddr == nil {
+			// We skip all modules without a source address
 			continue
 		}
 
-		moduleName := manifest.Key
-		subModuleName := ""
-
-		// determine if this module is nested in another module
-		// in the current workspace by finding a period in the moduleName
-		// is it better to look at SourceAddr and compare?
-		if strings.Contains(manifest.Key, ".") {
-			v := strings.Split(manifest.Key, ".")
-			moduleName = v[0]
-			subModuleName = v[1]
-		}
-
-		docsLink, err := getModuleDocumentationLink(ctx, manifest)
+		moduleName := manifest.LocalName
+		docsLink, err := getModuleDocumentationLink(ctx, manifest.SourceAddr.String(), manifest.Version.String())
```

Comment:
Nitpick: Is it worth making use of the interface resulting from an already parsed address here, instead of rebuilding the string and parsing it again within `getModuleDocumentationLink()`?